dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/put-fee-record-correction-transient-form-data.controller/index.test.ts

```diff
@@ -0,0 +1,199 @@
+import { ObjectId } from 'mongodb';
+import {
+  FeeRecordCorrectionTransientFormDataEntity,
+  RECORD_CORRECTION_REASON,
+  RecordCorrectionTransientFormData,
+  REQUEST_PLATFORM_TYPE,
+  TestApiError,
+} from '@ukef/dtfs2-common';
+import { HttpStatusCode } from 'axios';
+import { aTfmSessionUser } from '../../../../../../test-helpers';
+import { PutFeeRecordCorrectionTransientFormDataRequest, putFeeRecordCorrectionTransientFormData } from '.';
+import { PutFeeRecordCorrectionTransientFormDataPayload } from '../../../../routes/middleware/payload-validation';
+import { FeeRecordCorrectionTransientFormDataRepo } from '../../../../../repositories/fee-record-correction-transient-form-data-repo';
+import { FeeRecordRepo } from '../../../../../repositories/fee-record-repo';
+
+jest.mock('../../../../../repositories/fee-record-repo');
+jest.mock('../../../../../repositories/fee-record-correction-transient-form-data-repo');
+
+console.error = jest.fn();
+
+describe('put-fee-record-correction-transient-form-data.controller', () => {
+  describe('putFeeRecordCorrectionTransientFormData', () => {
+    const reportId = 1;
+    const feeRecordId = 2;
+
+    const tfmUserId = new ObjectId().toString();
+
+    const formData: RecordCorrectionTransientFormData = {
+      reasons: [RECORD_CORRECTION_REASON.FACILITY_ID_INCORRECT, RECORD_CORRECTION_REASON.OTHER],
+      additionalInfo: 'Some additional info',
+    };
+
+    const aValidRequestQuery = () => ({ reportId: reportId.toString(), feeRecordId: feeRecordId.toString() });
+
+    const aValidRequestBody = (): PutFeeRecordCorrectionTransientFormDataPayload => ({
+      user: { ...aTfmSessionUser(), _id: tfmUserId },
+      formData,
+    });
+
+    const mockFeeRecordExists = jest.fn();
+    const mockSave = jest.fn();
+
+    beforeEach(() => {
+      FeeRecordRepo.existsByIdAndReportId = mockFeeRecordExists;
+      mockFeeRecordExists.mockReturnValue(true);
+
+      FeeRecordCorrectionTransientFormDataRepo.save = mockSave;
+      mockSave.mockResolvedValue(undefined);
+    });
+
+    afterEach(() => {
+      jest.resetAllMocks();
+    });
+
+    it(`responds with a '${HttpStatusCode.Ok}' if the transient form data is saved successfully`, async () => {
+      // Arrange
+      const req = httpMocks.createRequest<PutFeeRecordCorrectionTransientFormDataRequest>({
+        params: aValidRequestQuery(),
+        body: aValidRequestBody(),
+      });
+      const res = httpMocks.createResponse();
+
+      const expectedTransientFormDataEntity = FeeRecordCorrectionTransientFormDataEntity.create({
+        userId: tfmUserId,
+        feeRecordId,
+        formData,
+        requestSource: {
+          platform: REQUEST_PLATFORM_TYPE.TFM,
+          userId: tfmUserId,
+        },
+      });
+
+      // Act
+      await putFeeRecordCorrectionTransientFormData(req, res);
+
+      // Assert
+      expect(res._getStatusCode()).toEqual(HttpStatusCode.Ok);
+
+      expect(mockFeeRecordExists).toHaveBeenCalledTimes(1);
```

Comment:
could these go in different it statements with a beforeEach? It is harder to follow if one of them fails for any reason.  As in one for the status code and another it for the toHaveBeenCalled